cli/deduplifhirLib/normalization.py

```diff
@@ -0,0 +1,337 @@
+    ' P.O. ': 'Post Office',
+    ' dept. ': 'department',
+    ' bldg. ': 'building',
+    ' rm. ': 'room',
+    ' ltd. ': 'limited',
+    ' co. ': 'company',
+    ' corp. ': 'corporation',
+    ' inc. ': 'incorporated',
+    ' intl. ': 'international',
+    ' u.k. ': 'United Kingdom',
+    ' u.s. ': 'United States',
+    ' apt. ': 'apartment',
+    ' ph. ': 'phone',
+    ' fax. ': 'fax',
+    ' mgr. ': 'manager',
+    ' asst. ': 'assistant',
+    ' dept ': 'department',
+    ' assoc. ': 'association',
+    ' edu. ': 'education',
+    ' univ. ': 'university',
+    ' inst. ': 'institute',
+    ' natl. ': 'national',
+    ' am. ': 'american',
+    ' eur. ': 'european',
+    ' intl ': 'international',
+    ' org. ': 'organization',
+    ' conf. ': 'conference',
+    ' symp. ': 'symposium',
+    ' AL ': 'Alabama',
+    ' AK ': 'Alaska',
+    ' AZ ': 'Arizona',
+    ' AR ': 'Arkansas',
+    ' CA ': 'California',
+    ' CO ': 'Colorado',
+    ' CT ': 'Connecticut',
+    ' DE ': 'Delaware',
+    ' FL ': 'Florida',
+    ' GA ': 'Georgia',
+    ' HI ': 'Hawaii',
+    ' ID ': 'Idaho',
+    ' IL ': 'Illinois',
+    ' IN ': 'Indiana',
+    ' IA ': 'Iowa',
+    ' KS ': 'Kansas',
+    ' KY ': 'Kentucky',
+    ' LA ': 'Louisiana',
+    ' ME ': 'Maine',
+    ' MD ': 'Maryland',
+    ' MA ': 'Massachusetts',
+    ' MI ': 'Michigan',
+    ' MN ': 'Minnesota',
+    ' MS ': 'Mississippi',
+    ' MO ': 'Missouri',
+    ' MT ': 'Montana',
+    ' NE ': 'Nebraska',
+    ' NV ': 'Nevada',
+    ' NH ': 'New Hampshire',
+    ' NJ ': 'New Jersey',
+    ' NM ': 'New Mexico',
+    ' NY ': 'New York',
+    ' NC ': 'North Carolina',
+    ' ND ': 'North Dakota',
+    ' OH ': 'Ohio',
+    ' OK ': 'Oklahoma',
+    ' OR ': 'Oregon',
+    ' PA ': 'Pennsylvania',
+    ' RI ': 'Rhode Island',
+    ' SC ': 'South Carolina',
+    ' SD ': 'South Dakota',
+    ' TN ': 'Tennessee',
+    ' TX ': 'Texas',
+    ' UT ': 'Utah',
+    ' VT ': 'Vermont',
+    ' VA ': 'Virginia',
+    ' WA ': 'Washington',
+    ' WV ': 'West Virginia',
+    ' WI ': 'Wisconsin',
+    ' WY ': 'Wyoming'
+}
+
+def add_permutations_to_dict_pattern(symbol_dict):
+    """
+    Add permutations of the keys from a dict and map them to the same value
+    as the original key. 
+
+    For example we make every key also match to its lowercase version,
+    as well as mapping to the version with punctuation and proper spaces. 
+
+
+    Arguments:
+        symbol_dict: dictionary of symbols to add permutations to 
+    
+    Returns:
+        Expanded version of dict input
+    """
+    #Add some permutations to the dictionary manually once
+    abvs = list(symbol_dict.keys())
+    for key in abvs:
+        symbol_dict[key.lower()] = symbol_dict[key]
+        symbol_dict[key[:-1] + ','] = symbol_dict[key]
+        symbol_dict[key[:-1] + '.'] = symbol_dict[key]
+
+
+        if key[-1] == '.':
+            symbol_dict[key[:-2] + ' '] = symbol_dict[key]
+            symbol_dict[key[:-2] + ', '] = symbol_dict[key]
```

Comment:
I think we can remove this now that we're removing non-alphanumeric unless I'm misunderstanding how that works?